### PR TITLE
🔥 Sled and use RocksDB instead

### PR DESCRIPTION
**Pull request by alexsnaps**
This builds on and supersedes both #146 & #167 … It replaces [sled](https://crates.io/crates/sled) as the underlying storage for local persistence with [RocksDB](https://rocksdb.org/) using the [Rust bindings](https://crates.io/crates/rocksdb)… which both were extensively battle tested. 

If we think this is indeed a better choice, it'd probably be worthwhile `git rebase -i` from HEAD of main, squashing all the work from the other PRs… I kept them as is for now, hoping this makes reviewing somewhat easier.

### CODE REVIEW

**Comment by eguzki** [comment]
> hoping this makes reviewing somewhat easier.

It does, thanks for that

**Comment by eguzki** [comment]
Looking good. I would go ahead and try to add the `disk persistence` feature, at least in `Tech Preview`

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
 ARG RUSTC_VERSION=1.67.1
 RUN apk update \
     && apk upgrade \
-    && apk add build-base binutils-gold openssl3-dev protoc protobuf-dev curl git \
+    && apk add build-base binutils-gold openssl3-dev protoc protobuf-dev curl git linux-headers clang \
```
Comment (eguzki) [inline Dockerfile:11]: there may be issues with the new ubi8 based images https://github.com/Kuadrant/limitador/pull/168

Comment (alexsnaps) [reply Dockerfile:11]: Totally will be...

**Comment by alexsnaps** [comment]
> Have you considered using the [Compaction Filter](https://github.com/facebook/rocksdb/wiki/RocksDB-Overview#compaction-filter) to implement the TTL??

I did, the issue is that 

> Have you considered using the [Compaction Filter](https://github.com/facebook/rocksdb/wiki/RocksDB-Overview#compaction-filter) to implement the TTL??

I do when disk optimized: https://github.com/Kuadrant/limitador/pull/170/files#diff-b41633d05540bf2734d97b29f9dab57f411c275100114fa933f6b0582cd04abaR124-R135

**Comment by alexsnaps** [comment]
Closed in favor of #178

**Pull request closed without merging**